commit ede3321166751d2cb49127d73d6c62c3388ffea7 (refs/changes/72/4672/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-08-13T14:39:20-07:00 (1 year, 2 months ago)
    
    TRITON-689 cn-agent's virtual servers for mockcloud don't set "transitional_status" which breaks 'sdcadm post-setup cmon'
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/CHANGES.md b/CHANGES.md
index c47e561..8933ab3 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,11 @@
 
 # sdcadm Changelog
 
+## 1.20.4
+
+- TRITON-689 improve `sdcadm post-setup cmon` to not require
+  `<server>.transitional_status` be defined on CNAPI server records.
+
 ## 1.20.3
 
 - TRITON-313 walk back sdcadm steps export sugar
diff --git a/lib/post-setup/cmon.js b/lib/post-setup/cmon.js
index 0fa9e6b..14cfbdb 100644
--- a/lib/post-setup/cmon.js
+++ b/lib/post-setup/cmon.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright 2018 Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 /*
  * `sdcadm post-setup cmon`
@@ -408,8 +408,7 @@ function do_cmon(subcmd, opts, args, cb) {
                 var notRunning = [];
                 ctx.serversToUpdate.forEach(function (srv) {
                     if (srv.status !== 'running' ||
-                        (srv.status === 'running' &&
-                         srv.transitional_status !== '')) {
+                        (srv.status === 'running' && srv.transitional_status)) {
                         notRunning.push(srv.uuid);
                     }
                 });
diff --git a/package.json b/package.json
index 582b564..a98d7df 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.20.3",
+  "version": "1.20.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
